assets/js/components/charts/Forecasts.jsx

```diff
@@ -118,16 +118,58 @@ export default class Forecasts extends Component<Props> {
       .x((d) => x(d.time))
       .y((d) => y(d.value))
 
+    var tooltip = d3
+      .select("body")
+      .append("div")
+      .classed("forecast-tooltip", true)
+      .style("visibility", "hidden")
+
+    for (var i = 0; i <= data.length; i++) {
+      if (data[i]) {
+        for (var j = 0; j <= data[i].values.length; j++) {
+          if (
+            data[i].values[j] &&
+            y(data[i].values[j].value) != 0 &&
+            x(data[i].values[j].time) != 0
+          ) {
+            d3.select(this.refs.circles)
+              .selectAll("path")
+              .data([data[i].values[j]])
+              .enter()
+              .append("circle")
+              .attr("cx", (d) => x(d.time))
+              .attr("cy", (d) => y(d.value))
+              .attr("r", "3px")
+              .style("fill", data[i].color)
+              .style("stroke", data[i].color)
+              .on("mouseover", (d) =>
+                tooltip
+                  .text(d.value)
+                  .style("top", d3.event.pageY - 10 + "px")
+                  .style("left", d3.event.pageX + 10 + "px")
+                  .style("visibility", "visible")
+              )
+              .on("mouseout", () => tooltip.style("visibility", "hidden"))
+          }
+        }
+      }
+    }
+
     d3.select(this.refs.values)
       .selectAll("path")
       .data(data)
       .enter()
       .append("path")
       .merge(d3.select(this.refs.values).selectAll("path"))
       .attr("class", "line")
+      .attr("id", (d) => "line-" + d.id)
       .attr("stroke", (d) => d.color)
       .datum((d) => d.values)
       .attr("d", line)
+      .on("click", function (d) {
+        d3.selectAll(".line").classed("clicked-line", false)
+        d3.selectAll("#" + d3.select(this).attr("id")).classed("clicked-line", true)
```

Comment:
Hi Julien! 

`d3.select(this).classed("clicked-line", true)` won't be enough because there is a line and a forecast with is its continuation that cannot be selected with `this`, that's why I added the id. 

Also, as I mentioned in the original comment, when extracting to a callback I get an error ( and also when making `(d)=>`) I will iterate on this to see what is the problem when extracting.

Reply:
The callback work, but I was doing it with `(d)=>` and this won't work... If I do

```
const lineClick = (d) => {
      d3.selectAll(".line").classed("clicked-line", false)
      d3.selectAll("#" + d3.select(this).attr("id")).classed("clicked-line", true)
    }
```

I will get: 

![image](https://user-images.githubusercontent.com/13782680/223181433-98a28044-2783-4f44-a3f8-25d32f2ca39e.png)

While with: 

```
    const lineClick = function (d) {
      d3.selectAll(".line").classed("clicked-line", false)
      d3.selectAll("#" + d3.select(this).attr("id")).classed("clicked-line", true)
    }
```

It will work

Reply:
Does that mean there are multiple elements with the same `#id`? That's not permitted by the DOM, though it looks like working in practice...

Reply:
About the duplicated ids, what about:

```js
.attr("class", `line line-${d.id}`)
.attr("data-line-id": d.id)
```

Reply:
I think the issue with the fat arrow (`=>`) is that `this` will be bound to the current `this` (i.e. the React component) when you want a callback where `this` is the actual element selected by D3. Using `function` is correct.